From 7c88a9334f371212d46fc6f62ca5d101ad431c55 Mon Sep 17 00:00:00 2001
From: Alexandre Julliard <julliard@winehq.org>
Date: Mon, 27 Oct 2025 16:28:48 +0100
Subject: [PATCH] ntdll: Don't map the PE header as executable.

Detecting noexec filesystems is not particularly useful since we fall
back to read()+mprotect() anyway.
---
 dlls/ntdll/unix/virtual.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 0bcd51a0f333..64ec42f9da73 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -1375,7 +1375,7 @@ static void dump_view( struct file_view *view )
     char *addr = view->base;
     BYTE prot = get_page_vprot( addr );
 
-    TRACE( "View: %p - %p", addr, addr + view->size - 1 );
+    TRACE( "View: %p - %p %s", addr, addr + view->size - 1, get_prot_str(view->protect) );
     if (view->protect & VPROT_SYSTEM)
         TRACE( " (builtin image)\n" );
     else if (view->protect & VPROT_FREE_PLACEHOLDER)
@@ -2745,8 +2745,7 @@ static NTSTATUS map_pe_header( void *ptr, size_t size, size_t map_size, int fd,
 
     if (!*removable && map_size)
     {
-        if (mmap( ptr, map_size, PROT_READ | PROT_WRITE | PROT_EXEC,
-                  MAP_FIXED | MAP_PRIVATE, fd, 0 ) != MAP_FAILED)
+        if (mmap( ptr, map_size, PROT_READ | PROT_WRITE, MAP_FIXED | MAP_PRIVATE, fd, 0 ) != MAP_FAILED)
         {
             if (size > map_size) pread( fd, (char *)ptr + map_size, size - map_size, map_size );
             return STATUS_SUCCESS;
